src/DepthImageToLaserScan.cpp

```diff
@@ -116,7 +116,7 @@ sensor_msgs::LaserScanPtr DepthImageToLaserScan::convert_msg(const sensor_msgs::
   scan_msg->range_max = range_max_;
   
   // Check scan_height vs image_height
-  if(scan_height_/2 > cam_model_.cy() || scan_height_/2 > depth_msg->height - cam_model_.cy()){
+  if(scan_height_/2 > (cam_model_.cy()*2*scan_offset_) || scan_height_/2 > depth_msg->height - (cam_model_.cy()*2*scan_offset_)){
```

Comment:
Should I make this into separate `if` ?
i.e. 
```
  // Check scan_height vs image_height
  if(scan_height_/2 > cam_model_.cy() || scan_height_/2 > depth_msg->height - cam_model_.cy()){
    std::stringstream ss;
    ss << "scan_height ( " << scan_height_ << " pixels) is too large for the image height.";
    throw std::runtime_error(ss.str());
  }

  // Check scan_height vs image_height and scan_offset
  if(scan_height_/2 > (cam_model_.cy()*2*scan_offset_) || scan_height_/2 > depth_msg->height - (cam_model_.cy()*2*scan_offset_)){
    std::stringstream ss;
    ss << "scan_height ( " << scan_height_ << " pixels) is too large for the current scan_offset.";
    throw std::runtime_error(ss.str());
  }
```

Reply:
Yes I thought it was OK but your change improves it. My comment was to suggest you create two auxiliary variables, one called `max_row` and one called `min_row` so that the ifs are cleaner.

Reply:
so what is the value of `max_row` and `min_row`?
are they like this?
```
max_row = scan_height_/2 > (cam_model_.cy()*2*scan_offset_)
min_row = scan_height_/2 > depth_msg->height - (cam_model_.cy()*2*scan_offset_)
```

Reply:
something like

    int centre_row = cam_model_.cy()*2*scan_offset_
    int bottom_row = centre_row + scan_height/2
    int top_row = centre_row - scan_height/2
    if(top_row < 0 || bottom_row >= depth_msg->height) {
        //ERROR
    }

This kind of clarity will make the code easier to understand and the pull request more easy to accept. You might want to consider making it elsewhere too.

Reply:
is it `center` or `centre`, Dr. Damien?

Also, In my logic I am sure the condition should be like this? Am I misunderstanding? Or you did it intentionally to check my awareness? :)
```
int centre_row = cam_model_.cy()*2*scan_offset_
int bottom_row = centre_row + scan_height/2
int top_row = centre_row - scan_height/2
if(bottom_row < 0 || top_row >= depth_msg->height) {
    //ERROR
}
```